### ISSUE TITLE
Problem with 3.0.1

### ISSUE THREAD

**Issue by LeDistordu** [nutso/redmine-plugin-recurring-tasks#69]
sudo rake redmine:plugins:migrate NAME=recurring_tasks RAILS_ENV=production --trace                                                                 [10:46:44]
(in /var/redmine)
[DEPRECATION] requiring "RMagick" is deprecated. Use "rmagick" instead
** Invoke redmine:plugins:migrate (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute redmine:plugins:migrate
Migrating recurring_tasks (Recurring Tasks (Issues))...
== 4 SetDefaultModifierForExistingMonthlyIssues: migrating ====================
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

undefined method `find_all_by_interval_unit' for #<Class:0x00000006bcedb8>/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/dynamic_matchers.rb:26:in `method_missing'
/var/redmine/plugins/recurring_tasks/db/migrate/004_set_default_modifier_for_existing_monthly_issues.rb:5:in `up'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:609:in `exec_migration'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/usr/lib/ruby/1.9.1/benchmark.rb:280:in `measure'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:589:in `block in migrate'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:588:in `migrate'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:765:in `migrate'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:1043:in `ddl_transaction'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:956:in `block in migrate'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `each'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `migrate'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:820:in `up'
/var/lib/gems/1.9.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:798:in `migrate'
/var/redmine/lib/redmine/plugin.rb:474:in `migrate_plugin'
/var/redmine/lib/redmine/plugin.rb:446:in `migrate'
/var/redmine/lib/redmine/plugin.rb:457:in `migrate'
/var/redmine/lib/tasks/redmine.rake:127:in `block (3 levels) in <top (required)>'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/lib/gems/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/var/lib/gems/1.9.1/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
Tasks: TOP => redmine:plugins:migrate

**Comment by insspb** [comment]
Hello.
Confirmed on another system. Same log on install. 
Master brunch without any modifications. 
```
 [redmine@redmine2 recurring_tasks]$ rake redmine:plugins:migrate RAILS_ENV=production --trace
(in /srv/redmine)
** Invoke redmine:plugins:migrate (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute redmine:plugins:migrate
Migrating clipboard_image_paste (Clipboard image paste)...
Migrating recurring_tasks (Recurring Tasks (Issues))...
== 4 SetDefaultModifierForExistingMonthlyIssues: migrating ====================
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

undefined method `find_all_by_interval_unit' for #<Class:0x00000007dea308>/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/dynamic_matchers.b:26:in `method_missing'
/srv/redmine/plugins/recurring_tasks/db/migrate/004_set_default_modifier_for_existing_monthly_issues.rb:5:in `up'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:609:in `exec_migration'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/2.0.0/benchmark.rb:281:in `measure'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:589:in `block in migrate'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:588:in `migrate'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:765:in `migrate'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:1043:in `ddl_transaction'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:956:in `block in migrate'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `each'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `migrate'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:820:in `up'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/gems/activerecord-4.2.1/lib/active_record/migration.rb:798:in `migrate'
/srv/redmine/lib/redmine/plugin.rb:474:in `migrate_plugin'
/srv/redmine/lib/redmine/plugin.rb:446:in `migrate'
/srv/redmine/lib/redmine/plugin.rb:460:in `block in migrate'
/srv/redmine/lib/redmine/plugin.rb:459:in `each'
/srv/redmine/lib/redmine/plugin.rb:459:in `migrate'
/srv/redmine/lib/tasks/redmine.rake:127:in `block (3 levels) in <top (required)>'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/srv/redmine/.rvm/rubies/ruby-2.0.0-p643/lib/ruby/gems/2.0.0/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/bin/rake:23:in `load'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/bin/rake:23:in `<main>'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/bin/ruby_executable_hooks:15:in `eval'
/srv/redmine/.rvm/gems/ruby-2.0.0-p643/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => redmine:plugins:migrate
[redmine@redmine2 recurring_tasks]$
```

**Comment by insspb** [comment]
I tried with develop branch. Another error:
```
[redmine@redmine2 recurring_tasks]$ rake redmine:plugins:migrate RAILS_ENV=production --trace
(in /srv/redmine)
An error occurred while loading the routes definition of recurring_tasks plugin (/srv/redmine/plugins/recurring_tasks/config/routes.rb): You should not use the `match` method in your router without specifying an HTTP method.
If you want to expose your action to both GET and POST, add `via: [:get, :post]` option.
If you want to expose your action to GET, use `get` in the router:
  Instead of: match "controller#action"
  Do: get "controller#action".
```

**Comment by gadgetsrule99** [comment]
I have not tried the development branch - any suggestions to solve this? @insspb did you have any luck solving the "undefined method `find_all_by_interval_unit'" issue?

**Comment by nutso** [comment]
The master branch should already resolve the routes issue as of a month or so ago. I just fixed the migration issue in develop, will be included in v1.5 after more testing  -- see #67 for notes.
